src/AggregateFunctions/QuantileReservoirSampler.h

```diff
@@ -65,11 +65,22 @@ struct QuantileReservoirSampler
     /// indices - an array of index levels such that the corresponding elements will go in ascending order.
     void getMany(const Float64 * levels, const size_t * indices, size_t size, Value * result)
     {
+        bool is_empty = data.empty();
+
         for (size_t i = 0; i < size; ++i)
-            if constexpr (is_decimal<Value>)
-                result[indices[i]] = Value(static_cast<typename Value::NativeType>(data.quantileInterpolated(levels[indices[i]])));
+        {
+            if (is_empty)
```

Comment:
Regarding UbSan fix, - this check is actually done inside `quantileInterpolated()` but this function on empty values, it returns NaN double. I fixed it in https://github.com/ClickHouse/ClickHouse/pull/43103, but it leads to changing behavior, see [here](https://github.com/ClickHouse/ClickHouse/pull/43103#issuecomment-1344159363).
If it's ok, then I guess, it'd be a preferable fix for undefined behavior:
(1) it respects onEmpty [policy](https://github.com/ClickHouse/ClickHouse/blob/06d6b87bae86f80c67633f0fe4f6d731cfd1cce7/src/AggregateFunctions/ReservoirSampler.h#L38) of `ReservoirSampler*` (not sure how it's important)
(2) it works for `QuantileReservoirSampler::get()` as well, current fix addresses only `getMany()`

Reply:
Changing the behavior on an empty column is not ok. Maybe not a big deal, but we are very concerned about incompatibilities recently.

I thought about removing the onEmpty policy, as after this change we have a duplication of logic, but did not get to that.

The point on `get` is valid, it should be fixed as well...